### PR TITLE
added full chat backend(db, api), added basic chat frontend

### PR DESCRIPTION
**Pull request by ajha7**

### CODE REVIEW

File: chat/__pycache__/chat_server.cpython-310.pyc
Comment (chennitoa) [inline chat/__pycache__/chat_server.cpython-310.pyc:1]: You need to add a gitignore in chat/ as well, make sure to ignore __pycache__ and anything else of significance

File: chat/requirements.txt
Comment (chennitoa) [inline chat/requirements.txt:1]: Make sure you only put what you need in the requirements file. You should make a virtual environment, manually install the packages again, and re-freeze.

**Review by chennitoa — CHANGES_REQUESTED** [review]
Make sure you remove the old chat folder.

File: package-lock.json
Comment (chennitoa) [inline package-lock.json:1]: Also remove these files, delete them entirely, they shouldnt be in the main folder.

**Comment by chennitoa** [comment]
You still need to delete package.json and package-lock.json. Also, you deleted .env.example, please restore that.

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,9 @@
-# Dotenv configurations
-**/.env
\ No newline at end of file
+**/.env
```
Comment (chennitoa) [inline .gitignore:1]: Don't mess with the main gitignore unless the file to be ignored should be in the root directory.

File: chat/chat_server.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
+
```
Comment (chennitoa) [inline chat/chat_server.py:1]: Use flake8 to format all of these files. Every python file.

**Comment by andrewkvu** [comment]
> You still need to delete package.json and package-lock.json. Also, you deleted .env.example, please restore that.

you'd still need to move whatevers in that into the normal package.json and package-lock inside the frontend folder no?

File: docker-compose.yaml [hunk 3]
```diff
@@ -35,6 +43,6 @@ services:
       - MYSQL_USER=${MYSQL_USER}
       - MYSQL_PASSWORD=${MYSQL_PASSWORD}
       - MYSQL_DATABASE=${MYSQL_DATABASE}
+      - MONGO_URI=${MONGO_URI}
```
Comment (chennitoa) [inline docker-compose.yaml]: does the mysql database use MONGO_URI? If not, remove this entry

**Review by chennitoa — CHANGES_REQUESTED** [review]
Fix everything. I will test tomorrow

**Comment by chennitoa** [comment]
> > You still need to delete package.json and package-lock.json. Also, you deleted .env.example, please restore that.
> 
> you'd still need to move whatevers in that into the normal package.json and package-lock inside the frontend folder no?

Those changes are made in the frontend/package.json and frontend/package-lock files

**Pull request merged**